### PR TITLE
ci: deprecate AppVoyer integration

### LINKED ISSUE #126: Proposal: drop AppVoyer in favor of GitHub Action
Is there any specific reason we still use AppVoyer? More often than not `continuous-integration/appveyor/branch` is stalled and blocks merging of, otherwise ready, PR.

In #87 we moved CI to GitHub action and continued it in #120 with additional workflow. In reality we can remove appvoyer integration as, afaict, it only runs the tests on Windows (which can be configured to be run on GH action anyways)

/cc @mavogel @clok

### PR DESCRIPTION
**Pull request by khos2ow**
## What does this do / why do we need it?

With the migration to GitHub Action, there's no need to continue to use
AppVoyer to only test on Windows, as it's covered by corresponding new
jobs.

## Check lists

* [x] Test passed
* [x] Coding style (indentation, etc)

## Which issue(s) does this PR fix?

Fixes #126

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -12,8 +12,10 @@ on:
       - 'README.md'
 jobs:
   unit:
-    runs-on: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
     strategy:
+      matrix:
+        os: [macos-latest, windows-latest]
```
Comment (khos2ow) [inline .github/workflows/test.yml:18]: For feature parity, I've added the matrix build here, but imo it's an overkill. We can reliably only use `unit` on `ubuntu-20.04` (or `ubuntu-latest` for that matter) which does also upload code coverage.

**Comment by clok** [comment]
This is interesting (and makes sense in a way), but the unit tests on windows fail due to carriage return instead of newline. See: https://github.com/git-chglog/git-chglog/runs/2177927696?check_suite_focus=true#step:4:50

File: .github/workflows/test.yml (hunk 1 above)
Comment (clok) [reply .github/workflows/test.yml:18]: This is really cool that this is supported so easily. It does feel like it is overkill at this point, but it is a "nice to have". I am open to either option really.

**Comment by clok** [comment]
So, do we update the test that is failing?

We could consider the OS when performing the expectation with the generated output. Or we could check that `\r\n` is in the output, then search/replace the output to perform the expectation check. Just some thoughts.

Or do we deprecate the windows test?

**Comment by clok** [comment]
PS - ❤️ the branch name

**Comment by khos2ow** [comment]
> Or we could check that \r\n is in the output, then search/replace the output to perform the expectation check.

That seems to be the issue, but this feels a bit weird to mutate the output based on the underlying OS. That said, I haven't had enough time since I opened this PR to delve into this. But my first feeling is that we should not have these.

https://github.com/git-chglog/git-chglog/blob/4d8b2b6be794b14fb959ea917f75eec974b79b14/chglog_test.go#L388

Although the difference betwen LF and CRLF might still be relevant to some folks.

**Comment by khos2ow** [comment]
So, it seems this issue is directly related to how Go manages line endings on different platforms. In other words `\n` works as is in console (even in Windows), the same goes for `fmt.Println`, but when it comes to read and write to the file Go will use LF or CRLF based on the underlying platform it's being executed on (https://github.com/golang/go/issues/28822)

That said, now, I'm in favor of always normalizing to LF. Because the intended destination of the output is a markdown file to be checked in a repository and be shared with broader audience. With that there won't be any difference between running `git-chglog` on a Linux or macOS or Windows.

**Comment by clok** [comment]
Nice find @khos2ow. After your explanation, it makes sense that we would be seeing this. 

I support normalizing to LF in our case. Your point about the desired destination of the tool output is spot on. 

I vote for deprecating the Windows tests at the least. I recognize that Mac OS is close enough to Linux in a general sense, so it may be overkill to include a separate Mac OS test run. That said, if it is minimal impact, it would be nice to keep the Mac OS test run. Mac is a common development platform.

**Comment by khos2ow** [comment]
> That said, now, I'm in favor of always normalizing to LF.

On the second though, I want to take back my earlier suggestion. I now think we'd best not mutate line endings at all and leave it to discretion of user to decide where they want to run this and how the line ending should look like. That's mainly because of the followings:

- it's computationally expensive (this has to be done line by line over a loop)
- the process deals with both read a file and write to a file
  - to expand on it, if it were only read and then output to stdout (i.e. no `io.Writer` involved) I might have considered it
- the choice is being completely removed from user and LF is being forced to them with no way to have CRLF

That said, and if we all agree to this, there won't be any need to run the tests on different platforms per se. They would behave correctly and accordingly on each of those as long as the tests are passing.

**Comment by clok** [comment]
> the choice is being completely removed from user and LF is being forced to them with no way to have CRLF

To me, that's the critical piece that makes me feel like this is something we don't need to worry about in these tests. 

If we had concerns about file permissions management or something that can differ drastically for each OS, I would have more concern.

**Review by khos2ow — COMMENTED** [review]
In light of the discussion above, I've reverted back matrix test and removed normalizing `s/\r\n/\n/` in tests (as it doesn't happen anywhere else in the code, and that seems a hack to make tests pass on Windows)

**Review by clok — APPROVED** [review]
Thank you for this. I have disabled the appvoyer checks on the mainline branch. See the image below. I will remove the webhook integration from the repo as well. Great work @khos2ow !

image

**Pull request merged**